WebCryptoAPI/digest/digest.js

```diff
@@ -0,0 +1,162 @@
+    var all_promises = [];
+
+    // Try every combination of hash with source data size. Variations tested are
+    // hash name in upper, lower, or mixed case, and upper-case version with the
+    // source buffer altered after call.
+    Object.keys(sourceData).forEach(function(size) {
+        Object.keys(digestedData).forEach(function(alg) {
+            var upCase = alg.toUpperCase();
+            var downCase = alg.toLowerCase();
+            var mixedCase = upCase.substr(0, 1) + downCase.substr(1);
+
+            promise_test(function(test) {
+                var promise = subtle.digest({name: upCase}, sourceData[size])
+                .then(function(result) {
+                    assert_true(equalBuffers(result, digestedData[alg][size]), "digest() yielded expected result for " + alg + ":" + size);
+                }, function(err) {
+                    assert_unreached("digest() threw an error for " + alg + ":" + size + " - " + err.message);
+                });
+
+                all_promises.push(promise);
+                return promise;
+            }, upCase + " with " + size + " source data");
+
+            promise_test(function(test) {
+                var promise = subtle.digest({name: mixedCase}, sourceData[size])
+                .then(function(result) {
+                    assert_true(equalBuffers(result, digestedData[alg][size]), "digest() yielded expected result for " + alg + ":" + size);
+                }, function(err) {
+                    assert_unreached("digest() threw an error for " + alg + ":" + size + " - " + err.message);mixedCase
+                });
+
+                all_promises.push(promise);
+                return promise;
+            }, downCase + " with " + size + " source data");
+
+            promise_test(function(test) {
+                var promise = subtle.digest({name: mixedCase}, sourceData[size])
+                .then(function(result) {
+                    assert_true(equalBuffers(result, digestedData[alg][size]), "digest() yielded expected result for " + alg + ":" + size);
+                }, function(err) {
+                    assert_unreached("digest() threw an error for " + alg + ":" + size + " - " + err.message);
+                });
+
+                all_promises.push(promise);
+                return promise;
+            }, mixedCase + " with " + size + " source data");
+
+            promise_test(function(test) {
+                var copiedBuffer = copyBuffer(sourceData[size]);
+                var promise = subtle.digest({name: upCase}, copiedBuffer)
+                .then(function(result) {
+                    copiedBuffer[0] = 255 - copiedBuffer;
+                    assert_true(equalBuffers(result, digestedData[alg][size]), "digest() yielded expected result for " + alg + ":" + size);
+                }, function(err) {
+                    assert_unreached("digest() threw an error for " + alg + ":" + size + " - " + err.message);
+                });
+
```

Comment:
I can't tell where you mean, but I think I understand the issue and have
fixed it. I've also changed done() handling.

On Wed, Jun 8, 2016 at 7:44 PM, Jim Schaad <notifications@github.com> wrote:

> In WebCryptoAPI/digest/digest.js
> <https://github.com/w3c/web-platform-tests/pull/3132#discussion_r66361324>
> :
>
> > +                });
> > +
> > +                all_promises.push(promise);
> > +                return promise;
> > +            }, mixedCase + " with " + size + " source data");
> > +
> > +            promise_test(function(test) {
> > +                var copiedBuffer = copyBuffer(sourceData[size]);
> > +                var promise = subtle.digest({name: upCase}, copiedBuffer)
> > +                .then(function(result) {
> > +                    copiedBuffer[0] = 255 - copiedBuffer;
> > +                    assert_true(equalBuffers(result, digestedData[alg][size]), "digest() yielded expected result for " + alg + ":" + size);
> > +                }, function(err) {
> > +                    assert_unreached("digest() threw an error for " + alg + ":" + size + " - " + err.message);
> > +                });
> > +
>
> copiedBuffer modification goes here - it needs to be after the digest has
> started and before it has finished.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/w3c/web-platform-tests/pull/3132/files/04c7efe5afe976935d5be5798c19dc3547f2b3eb#r66361324>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAE_N2absGGwHUpV8_i1er-6L6Qlm1weks5qJ1PLgaJpZM4IxXWL>
> .
>